diff -aur org/target/icu/source/config/Makefile.inc.in target/icu/source/config/Makefile.inc.in
--- org/target/icu/source/config/Makefile.inc.in	2017-11-20 17:16:30.000000000 +0100
+++ target/icu/source/config/Makefile.inc.in	2017-09-22 15:58:41.225928503 +0200
@@ -142,7 +142,7 @@
 
 # Versioned target for a shared library
 FINAL_SO_TARGET = $(SO_TARGET).$(SO_TARGET_VERSION)
-MIDDLE_SO_TARGET = $(SO_TARGET).$(SO_TARGET_VERSION_MAJOR)
+MIDDLE_SO_TARGET = $(SO_TARGET)
 
 # Access to important ICU tools.
 # Use as follows:  $(INVOKE) $(GENRB) arguments .. 
diff -aur org/target/icu/source/config/mh-linux target/icu/source/config/mh-linux
--- org/target/icu/source/config/mh-linux	2017-11-20 17:16:30.000000000 +0100
+++ target/icu/source/config/mh-linux	2017-09-22 15:58:41.225928503 +0200
@@ -27,9 +27,12 @@
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
-	LD_SONAME = -Wl,-soname -Wl,$(notdir $(MIDDLE_SO_TARGET))
+	LD_SONAME = -Wl,-soname -Wl,$(notdir $(SO_TARGET))
+	COMMON_STUBNAME = uc_jsc
+	I18N_STUBNAME = i18n_jsc
+	DATA_STUBNAME = data_jsc
 #SH# # We can't depend on MIDDLE_SO_TARGET being set.
-#SH# LD_SONAME=
+#SH# LD_SONAME=${SO_TARGET}
 
 ## Shared library options
 LD_SOOPTIONS= -Wl,-Bsymbolic
diff -aur org/target/icu/source/data/Makefile.in target/icu/source/data/Makefile.in
--- org/target/icu/source/data/Makefile.in	2017-11-20 17:16:30.000000000 +0100
+++ target/icu/source/data/Makefile.in	2017-09-22 15:58:41.225928503 +0200
@@ -26,9 +26,9 @@
 ifeq ($(PKGDATA_OPTS),)
 PKGDATA_OPTS = -O $(top_builddir)/data/icupkg.inc
 endif
-ifeq ($(PKGDATA_VERSIONING),)
-PKGDATA_VERSIONING = -r $(SO_TARGET_VERSION)
-endif
+#ifeq ($(PKGDATA_VERSIONING),)
+#PKGDATA_VERSIONING = -r $(SO_TARGET_VERSION)
+#endif
 
 # This allows all the data to be in one directory
 ifeq ($(PKGDATA_MODE),dll)
@@ -452,6 +452,11 @@
 MISC_FILES_LIST=$(DAT_FILES_SHORT) $(CNV_FILES_SHORT) $(CNV_FILES_SHORT_SPECIAL) $(CURR_FILES_SHORT) $(RBNF_FILES_SHORT) $(RBNF_INDEX_RES_SHORT) $(TRANSLIT_FILES_SHORT) $(SPREP_FILES_SHORT) $(CFU_FILES_SHORT)
 UNI_CORE_DATA=pnames.icu uprops.icu ucase.icu ubidi.icu nfc.nrm
 UNI_CORE_TARGET_DATA=$(UNI_CORE_DATA:%=$(BUILDDIR)/%)
+BRK_FILES_LIST=""
+LOCALE_FILES_LIST=$(RES_FILES_SHORT) 
+MISC_FILES_LIST=coll/ucadata.icu
+
+
 
 ifneq ($(INCLUDE_UNI_CORE_DATA),)
 MISC_FILES_LIST+=$(UNI_CORE_DATA)
