diff --exclude=.svn -aur icu/source/config/Makefile.inc.in target/icu/source/config/Makefile.inc.in
--- icu/source/config/Makefile.inc.in	2017-05-22 07:24:35.000000000 +0200
+++ target/icu/source/config/Makefile.inc.in	2017-05-16 16:03:46.000000000 +0200
@@ -143,7 +143,7 @@
 
 # Versioned target for a shared library
 FINAL_SO_TARGET = $(SO_TARGET).$(SO_TARGET_VERSION)
-MIDDLE_SO_TARGET = $(SO_TARGET).$(SO_TARGET_VERSION_MAJOR)
+MIDDLE_SO_TARGET = $(SO_TARGET)
 
 # Access to important ICU tools.
 # Use as follows:  $(INVOKE) $(GENRB) arguments .. 
diff --exclude=.svn -aur icu/source/config/mh-linux target/icu/source/config/mh-linux
--- icu/source/config/mh-linux	2017-05-22 07:24:35.000000000 +0200
+++ target/icu/source/config/mh-linux	2017-05-19 16:16:50.000000000 +0200
@@ -25,9 +25,12 @@
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
-	LD_SONAME = -Wl,-soname -Wl,$(notdir $(MIDDLE_SO_TARGET))
+	LD_SONAME = -Wl,-soname -Wl,$(notdir $(SO_TARGET))
+	COMMON_STUBNAME = uc_jsc
+	I18N_STUBNAME = i18n_jsc
+	DATA_STUBNAME = data_jsc
 #SH# # We can't depend on MIDDLE_SO_TARGET being set.
-#SH# LD_SONAME=
+#SH# LD_SONAME=${SO_TARGET}
 
 ## Shared library options
 LD_SOOPTIONS= -Wl,-Bsymbolic
