diff -aur target-org/webkit/Source/bmalloc/bmalloc/DebugHeap.cpp target/webkit/Source/bmalloc/bmalloc/DebugHeap.cpp
--- target-org/webkit/Source/bmalloc/bmalloc/DebugHeap.cpp      2017-02-03 22:59:08.000000000 +0100
+++ target/webkit/Source/bmalloc/bmalloc/DebugHeap.cpp  2017-08-02 10:07:11.000383124 +0200
@@ -84,13 +84,8 @@

 void* DebugHeap::memalign(size_t alignment, size_t size, bool crashOnFailure)
 {
-    void* result;
-    if (posix_memalign(&result, alignment, size)) {
-        if (crashOnFailure)
-            BCRASH();
-        return nullptr;
-    }
-    return result;
+    BCRASH();
+    return nullptr;
 }

 void* DebugHeap::realloc(void* object, size_t size)
diff -aur target-org/webkit/Source/cmake/FindICU.cmake target/webkit/Source/cmake/FindICU.cmake
--- target-org/webkit/Source/cmake/FindICU.cmake	2016-11-30 19:45:57.000000000 +0100
+++ target/webkit/Source/cmake/FindICU.cmake	2017-07-11 11:34:55.962374878 +0200
@@ -25,7 +25,7 @@
 # Look for the library.
 find_library(
     ICU_LIBRARY
-    NAMES libicuuc cygicuuc cygicuuc32 icuuc
+    NAMES libicuuc cygicuuc cygicuuc32 icuuc icuuc_jsc
     HINTS ${PC_ICU_LIBRARY_DIRS}
           ${PC_ICU_LIBDIR}
           ${WEBKIT_LIBRARIES_LINK_DIR}
@@ -51,7 +51,7 @@
     pkg_check_modules(PC_ICU_I18N icu-i18n)
     find_library(
         ICU_I18N_LIBRARY
-        NAMES icui18n libicui18n libicuin cygicuin cygicuin32 icuin
+        NAMES icui18n libicui18n libicuin cygicuin cygicuin32 icuin icui18n_jsc
         HINTS ${PC_ICU_I18N_LIBRARY_DIRS}
               ${PC_ICU_I18N_LIBDIR}
               ${WEBKIT_LIBRARIES_LINK_DIR}
@@ -68,7 +68,7 @@
     # Look for the ICU data libraries
     find_library(
         ICU_DATA_LIBRARY
-        NAMES icudata libicudata cygicudata cygicudata32
+        NAMES icudata libicudata cygicudata cygicudata32 icudata_jsc
         HINTS ${PC_ICU_I18N_LIBRARY_DIRS}
               ${PC_ICU_I18N_LIBDIR}
               ${WEBKIT_LIBRARIES_LINK_DIR}
diff -aur target-org/webkit/Source/cmake/OptionsJSCOnly.cmake target/webkit/Source/cmake/OptionsJSCOnly.cmake
--- target-org/webkit/Source/cmake/OptionsJSCOnly.cmake	2017-04-17 15:51:21.000000000 +0200
+++ target/webkit/Source/cmake/OptionsJSCOnly.cmake	2017-07-11 11:34:55.962374878 +0200
@@ -34,12 +34,11 @@
 set(ENABLE_WEBCORE OFF)
 set(ENABLE_WEBKIT_LEGACY OFF)
 set(ENABLE_WEBKIT OFF)
-
-if (WIN32)
-    set(ENABLE_API_TESTS OFF)
-else ()
-    set(ENABLE_API_TESTS ON)
-endif ()
+set(ENABLE_API_TESTS OFF)
+set(ENABLE_SAMPLING_PROFILER OFF)
+set(ENABLE_DFG_JIT OFF)
+set(ENABLE_FTL_JIT OFF)
+set(ENABLE_JIT ON)

 if (WTF_CPU_X86 OR WTF_CPU_X86_64)
     SET_AND_EXPOSE_TO_BUILD(USE_UDIS86 1)
diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/Source/JavaScriptCore/CMakeLists.txt
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2017-05-02 21:13:03.000000000 +0200
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2017-07-11 11:34:55.962374878 +0200
@@ -1543,7 +1543,7 @@

 target_include_directories(LLIntOffsetsExtractor PRIVATE ${JavaScriptCore_INCLUDE_DIRECTORIES})

-add_subdirectory(shell)
+# add_subdirectory(shell)

 WEBKIT_WRAP_SOURCELIST(${JavaScriptCore_SOURCES})
 WEBKIT_FRAMEWORK(JavaScriptCore)
@@ -1555,6 +1555,7 @@
         install(TARGETS JavaScriptCore DESTINATION "${LIB_INSTALL_DIR}")
     endif ()
 endif ()
+set_target_properties(JavaScriptCore PROPERTIES OUTPUT_NAME jsc)

 # Force staging of shared scripts, even if they aren't directly used to build JavaScriptCore.

diff -aur target-org/webkit/Source/WTF/wtf/PlatformUserPreferredLanguagesUnix.cpp target/webkit/Source/WTF/wtf/PlatformUserPreferredLanguagesUnix.cpp
--- target-org/webkit/Source/WTF/wtf/PlatformUserPreferredLanguagesUnix.cpp	2016-10-08 05:31:15.000000000 +0200
+++ target/webkit/Source/WTF/wtf/PlatformUserPreferredLanguagesUnix.cpp	2017-07-11 11:34:55.962374878 +0200
@@ -24,24 +24,29 @@
 #include <locale.h>
 #include <wtf/Vector.h>
 #include <wtf/text/WTFString.h>
+#include <sys/system_properties.h>

 namespace WTF {

 void setPlatformUserPreferredLanguagesChangedCallback(void (*)()) { }

-// Using pango_language_get_default() here is not an option, because
-// it doesn't support changing the locale in runtime, so it returns
-// always the same value.
 static String platformLanguage()
 {
-    String localeDefault(setlocale(LC_CTYPE, nullptr));
-    if (localeDefault.isEmpty() || equalIgnoringASCIICase(localeDefault, "C") || equalIgnoringASCIICase(localeDefault, "POSIX"))
+    char buf[2 * (PROP_VALUE_MAX + 1)];
+
+    int locale_len = __system_property_get("persist.sys.locale", buf);
+    if (locale_len == 0) {
+      int lang_len = __system_property_get("persist.sys.language", buf);
+      if(lang_len == 0) {
         return ASCIILiteral("en-US");
+      }
+      int country_len = __system_property_get("persist.sys.country", buf + lang_len + 1);
+      if(country_len != 0) {
+        buf[lang_len] = '-';
+      }
+    }

-    String normalizedDefault = localeDefault;
-    normalizedDefault.replace('_', '-');
-    normalizedDefault.truncate(normalizedDefault.find('.'));
-    return normalizedDefault;
+    return String(buf);
 }

 Vector<String> platformUserPreferredLanguages()
diff -aur target-org/webkit/Source/WTF/wtf/StackTrace.cpp target/webkit/Source/WTF/wtf/StackTrace.cpp
--- target-org/webkit/Source/WTF/wtf/StackTrace.cpp	2017-04-25 04:53:49.000000000 +0200
+++ target/webkit/Source/WTF/wtf/StackTrace.cpp	2017-07-11 11:34:55.962374878 +0200
@@ -30,10 +30,6 @@
 #include <wtf/Assertions.h>
 #include <wtf/PrintStream.h>

-#if HAVE(BACKTRACE_SYMBOLS) || HAVE(BACKTRACE)
-#include <execinfo.h>
-#endif
-
 #if HAVE(DLADDR)
 #include <cxxabi.h>
 #include <dlfcn.h>
diff -aur target-org/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp target/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp
--- target-org/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp	2014-03-15 05:08:27.000000000 +0100
+++ target/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp	2017-07-11 11:36:01.845264855 +0200
@@ -28,12 +28,13 @@

 #include "config.h"
 #include "Collator.h"
+#include <StringView.h>

 #if UCONFIG_NO_COLLATION

 namespace WTF {

-int Collator::collate(StringView a, StringView b) const
+int Collator::collate(StringView a, StringView b)
 {
     unsigned commonLength = std::min(a.length(), b.length());
     for (unsigned i = 0; i < commonLength; ++i) {
@@ -51,7 +52,7 @@
     return 0;
 }

-int Collator::collateUTF8(const char* a, const char* b) const
+int Collator::collateUTF8(const char* a, const char* b)
 {
     return collate(String::fromUTF8(a), String::fromUTF8(b));
 }
diff -aur org/target/webkit/Source/cmake/OptionsCommon.cmake target/webkit/Source/cmake/OptionsCommon.cmake
--- target-org/webkit/Source/cmake/OptionsCommon.cmake      2017-11-15 16:28:52.120093206 +0100
+++ target/webkit/Source/cmake/OptionsCommon.cmake  2017-04-09 02:50:32.875166000 +0200
@@ -255,7 +255,7 @@

 # Check for headers
 _HAVE_CHECK_INCLUDE(HAVE_ERRNO_H errno.h)
-_HAVE_CHECK_INCLUDE(HAVE_LANGINFO_H langinfo.h)
+#_HAVE_CHECK_INCLUDE(HAVE_LANGINFO_H langinfo.h)
 _HAVE_CHECK_INCLUDE(HAVE_MMAP sys/mman.h)
 _HAVE_CHECK_INCLUDE(HAVE_PTHREAD_NP_H pthread_np.h)
 _HAVE_CHECK_INCLUDE(HAVE_STRINGS_H strings.h)
diff -aur target-org/webkit/Source/cmake/FindICU.cmake target/webkit/Source/cmake/FindICU.cmake
--- target-org/webkit/Source/cmake/FindICU.cmake	2018-07-24 15:41:23.452474811 +0200
+++ target/webkit/Source/cmake/FindICU.cmake	2018-07-12 15:53:44.611542333 +0200
@@ -16,19 +16,15 @@
 find_path(
     ICU_INCLUDE_DIR
     NAMES unicode/utypes.h
-    HINTS ${PC_ICU_INCLUDE_DIRS}
-          ${PC_ICU_INCLUDEDIR}
-          ${WEBKIT_LIBRARIES_INCLUDE_DIR}
+    HINTS ${WEBKIT_LIBRARIES_INCLUDE_DIR}
     DOC "Include directory for the ICU library")
 mark_as_advanced(ICU_INCLUDE_DIR)
 
 # Look for the library.
 find_library(
     ICU_LIBRARY
-    NAMES libicuuc cygicuuc cygicuuc32 icuuc icuuc_jsc
-    HINTS ${PC_ICU_LIBRARY_DIRS}
-          ${PC_ICU_LIBDIR}
-          ${WEBKIT_LIBRARIES_LINK_DIR}
+    NAMES icuuc_jsc
+    HINTS ${WEBKIT_LIBRARIES_LINK_DIR}
     DOC "Libraries to link against for the common parts of ICU")
 mark_as_advanced(ICU_LIBRARY)
 
@@ -51,10 +47,8 @@
     pkg_check_modules(PC_ICU_I18N icu-i18n)
     find_library(
         ICU_I18N_LIBRARY
-        NAMES icui18n libicui18n libicuin cygicuin cygicuin32 icuin icui18n_jsc
-        HINTS ${PC_ICU_I18N_LIBRARY_DIRS}
-              ${PC_ICU_I18N_LIBDIR}
-              ${WEBKIT_LIBRARIES_LINK_DIR}
+        NAMES icui18n_jsc
+        HINTS ${WEBKIT_LIBRARIES_LINK_DIR}
         DOC "Libraries to link against for ICU internationalization")
     mark_as_advanced(ICU_I18N_LIBRARY)
     if (ICU_I18N_LIBRARY)
@@ -68,10 +62,8 @@
     # Look for the ICU data libraries
     find_library(
         ICU_DATA_LIBRARY
-        NAMES icudata libicudata cygicudata cygicudata32 icudata_jsc
-        HINTS ${PC_ICU_I18N_LIBRARY_DIRS}
-              ${PC_ICU_I18N_LIBDIR}
-              ${WEBKIT_LIBRARIES_LINK_DIR}
+        NAMES icudata_jsc
+        HINTS ${WEBKIT_LIBRARIES_LINK_DIR}
         DOC "Libraries to link against for ICU data")
     mark_as_advanced(ICU_DATA_LIBRARY)
     if (ICU_DATA_LIBRARY)
diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/Source/JavaScriptCore/CMakeLists.txt
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2018-07-25 09:56:23.662494914 +0200
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2018-07-18 12:55:24.726736260 +0200
@@ -1072,6 +1072,8 @@
 set(JavaScriptCore_LIBRARIES
     WTF${DEBUG_SUFFIX}
     ${ICU_I18N_LIBRARIES}
+    ${ICU_LIBRARIES}
+    ${ICU_DATA_LIBRARIES}
     ${LLVM_LIBRARIES}
 )
 
diff -aur target-org/webkit/Source/JavaScriptCore/API/JSBase.cpp target/webkit/Source/JavaScriptCore/API/JSBase.cpp
--- target-org/webkit/Source/JavaScriptCore/API/JSBase.cpp	2017-05-05 02:37:52.000000000 +0300
+++ target/webkit/Source/JavaScriptCore/API/JSBase.cpp	2018-10-09 14:38:30.000000000 +0300
@@ -191,3 +191,9 @@
 const char iosInstallName60 = 0;
 const char iosInstallName61 = 0;
 #endif
+
+extern "C" int __android_log_print(int prio, const char *tag, const char *fmt, ...);
+__attribute__((constructor))
+void printVersion() {
+    __android_log_print(3, "JavaScriptCore-Version", "%s", "224109.1.0");
+}
diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/Source/JavaScriptCore/CMakeLists.txt
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2017-10-16 15:19:26.000000000 +0300
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2018-10-09 14:44:23.000000000 +0300
@@ -1073,6 +1073,7 @@
     WTF${DEBUG_SUFFIX}
     ${ICU_I18N_LIBRARIES}
     ${LLVM_LIBRARIES}
+    log
 )
 
 set(JavaScriptCore_SCRIPTS_SOURCES_DIR "${JAVASCRIPTCORE_DIR}/Scripts")
