diff -aur orig/webkit/Source/cmake/FindICU.cmake webkit/Source/cmake/FindICU.cmake
--- orig/webkit/Source/cmake/FindICU.cmake	2016-11-30 19:45:57.627237000 +0100
+++ webkit/Source/cmake/FindICU.cmake	2017-06-14 10:45:36.042729515 +0200
@@ -25,7 +25,7 @@
 # Look for the library.
 find_library(
     ICU_LIBRARY
-    NAMES libicuuc cygicuuc cygicuuc32 icuuc
+    NAMES libicuuc cygicuuc cygicuuc32 icuuc icuuc_jsc
     HINTS ${PC_ICU_LIBRARY_DIRS}
           ${PC_ICU_LIBDIR}
           ${WEBKIT_LIBRARIES_LINK_DIR}
@@ -51,7 +51,7 @@
     pkg_check_modules(PC_ICU_I18N icu-i18n)
     find_library(
         ICU_I18N_LIBRARY
-        NAMES icui18n libicui18n libicuin cygicuin cygicuin32 icuin
+        NAMES icui18n libicui18n libicuin cygicuin cygicuin32 icuin icui18n_jsc
         HINTS ${PC_ICU_I18N_LIBRARY_DIRS}
               ${PC_ICU_I18N_LIBDIR}
               ${WEBKIT_LIBRARIES_LINK_DIR}
@@ -68,7 +68,7 @@
     # Look for the ICU data libraries
     find_library(
         ICU_DATA_LIBRARY
-        NAMES icudata libicudata cygicudata cygicudata32
+        NAMES icudata libicudata cygicudata cygicudata32 icudata_jsc
         HINTS ${PC_ICU_I18N_LIBRARY_DIRS}
               ${PC_ICU_I18N_LIBDIR}
               ${WEBKIT_LIBRARIES_LINK_DIR}
diff -aur orig/webkit/Source/cmake/OptionsJSCOnly.cmake webkit/Source/cmake/OptionsJSCOnly.cmake
--- orig/webkit/Source/cmake/OptionsJSCOnly.cmake	2017-04-17 15:51:21.809161000 +0200
+++ webkit/Source/cmake/OptionsJSCOnly.cmake	2017-06-14 10:45:36.042729515 +0200
@@ -23,7 +23,8 @@
 set(ENABLE_WEBCORE OFF)
 set(ENABLE_WEBKIT OFF)
 set(ENABLE_WEBKIT2 OFF)
-set(ENABLE_API_TESTS ON)
+set(ENABLE_API_TESTS OFF)
+set(ENABLE_SAMPLING_PROFILER OFF)
 
 if (WTF_CPU_X86 OR WTF_CPU_X86_64)
     SET_AND_EXPOSE_TO_BUILD(USE_UDIS86 1)
diff -aur orig/webkit/Source/JavaScriptCore/CMakeLists.txt webkit/Source/JavaScriptCore/CMakeLists.txt
--- orig/webkit/Source/JavaScriptCore/CMakeLists.txt	2017-05-02 21:13:03.684637000 +0200
+++ webkit/Source/JavaScriptCore/CMakeLists.txt	2017-06-14 10:45:36.042729515 +0200
@@ -1543,7 +1543,7 @@
 
 target_include_directories(LLIntOffsetsExtractor PRIVATE ${JavaScriptCore_INCLUDE_DIRECTORIES})
 
-add_subdirectory(shell)
+# add_subdirectory(shell)
 
 WEBKIT_WRAP_SOURCELIST(${JavaScriptCore_SOURCES})
 WEBKIT_FRAMEWORK(JavaScriptCore)
@@ -1555,6 +1555,7 @@
         install(TARGETS JavaScriptCore DESTINATION "${LIB_INSTALL_DIR}")
     endif ()
 endif ()
+set_target_properties(JavaScriptCore PROPERTIES OUTPUT_NAME jsc)
 
 # Force staging of shared scripts, even if they aren't directly used to build JavaScriptCore.
 
diff -aur orig/webkit/Source/WTF/wtf/Assertions.cpp webkit/Source/WTF/wtf/Assertions.cpp
--- orig/webkit/Source/WTF/wtf/Assertions.cpp	2017-04-25 04:53:49.356899000 +0200
+++ webkit/Source/WTF/wtf/Assertions.cpp	2017-06-14 11:09:30.575649644 +0200
@@ -74,7 +74,6 @@
 #endif
 
 #if HAVE(BACKTRACE)
-#include <execinfo.h>
 #endif
 
 extern "C" {
diff -aur orig/webkit/Source/WTF/wtf/StackTrace.cpp webkit/Source/WTF/wtf/StackTrace.cpp
--- orig/webkit/Source/WTF/wtf/StackTrace.cpp	2017-04-25 04:53:49.356899000 +0200
+++ webkit/Source/WTF/wtf/StackTrace.cpp	2017-06-14 11:09:10.912006447 +0200
@@ -33,7 +33,6 @@
 #if HAVE(BACKTRACE_SYMBOLS) || HAVE(DLADDR)
 #include <cxxabi.h>
 #include <dlfcn.h>
-#include <execinfo.h>
 #endif
 
 namespace WTF {
