diff -aur target-org/webkit/Source/JavaScriptCore/API/JSBase.cpp target/webkit/Source/JavaScriptCore/API/JSBase.cpp
--- target-org/webkit/Source/JavaScriptCore/API/JSBase.cpp	2017-05-05 02:37:52.000000000 +0300
+++ target/webkit/Source/JavaScriptCore/API/JSBase.cpp	2018-10-09 14:38:30.000000000 +0300
@@ -197,3 +197,9 @@
 const char iosInstallName60 = 0;
 const char iosInstallName61 = 0;
 #endif
+
+extern "C" int __android_log_print(int prio, const char *tag, const char *fmt, ...);
+__attribute__((constructor))
+void printVersion() {
+    __android_log_print(3, "JavaScriptCore.Version", "%s", JSC_VERSION);
+}
diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/Source/JavaScriptCore/CMakeLists.txt
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2019-09-17 13:57:03.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2019-09-17 13:57:08.000000000 +0800
@@ -120,6 +120,9 @@

 list(APPEND JavaScriptCore_LIBRARIES ${ICU_LIBRARIES} ${ICU_DATA_LIBRARIES})

+list(APPEND JavaScriptCore_LIBRARIES log)
+add_definitions(-DJSC_VERSION="${JSC_VERSION}")
+
 if (USE_CAPSTONE)
     list(APPEND JavaScriptCore_LIBRARIES capstone)
 endif ()
