diff -aur target-org/webkit/Source/WTF/wtf/StackTrace.cpp target/webkit/Source/WTF/wtf/StackTrace.cpp
--- target-org/webkit/Source/WTF/wtf/StackTrace.cpp	2017-04-25 04:53:49.000000000 +0200
+++ target/webkit/Source/WTF/wtf/StackTrace.cpp	2017-07-11 11:34:55.962374878 +0200
@@ -30,10 +30,6 @@
 #include <wtf/Assertions.h>
 #include <wtf/PrintStream.h>

-#if HAVE(BACKTRACE_SYMBOLS) || HAVE(BACKTRACE)
-#include <execinfo.h>
-#endif
-
 #if HAVE(DLADDR)
 #include <cxxabi.h>
 #include <dlfcn.h>
