diff --git a/webkit/Source/JavaScriptCore/jit/JITInlineCacheGenerator.h b/webkit/Source/JavaScriptCore/jit/JITInlineCacheGenerator.h
index 8b89733b83..9c1d93c23f 100644
--- a/webkit/Source/JavaScriptCore/jit/JITInlineCacheGenerator.h
+++ b/webkit/Source/JavaScriptCore/jit/JITInlineCacheGenerator.h
@@ -53,7 +53,11 @@ enum class AccessType : int8_t;
 enum class CacheType : int8_t;
 enum class JITType : uint8_t;
 
+#if ENABLE(DFG_JIT)
 using CompileTimeStructureStubInfo = Variant<StructureStubInfo*, BaselineUnlinkedStructureStubInfo*, DFG::UnlinkedStructureStubInfo*>;
+#else
+using CompileTimeStructureStubInfo = Variant<StructureStubInfo*, BaselineUnlinkedStructureStubInfo*>;
+#endif
 
 class JITInlineCacheGenerator {
 protected:
diff --git a/webkit/Source/JavaScriptCore/jit/PCToCodeOriginMap.h b/webkit/Source/JavaScriptCore/jit/PCToCodeOriginMap.h
index 196e0516a9..572f6a4444 100644
--- a/webkit/Source/JavaScriptCore/jit/PCToCodeOriginMap.h
+++ b/webkit/Source/JavaScriptCore/jit/PCToCodeOriginMap.h
@@ -41,6 +41,7 @@ namespace B3 {
 class PCToOriginMap;
 }
 
+#if ENABLE(WEBASSEMBLY)
 namespace Wasm {
 class WasmOrigin {
     MAKE_VALIDATED_REINTERPRET_CAST
@@ -59,6 +60,7 @@ public:
 MAKE_VALIDATED_REINTERPRET_CAST_IMPL("WasmOrigin", WasmOrigin)
 
 } // namespace Wasm
+#endif
 
 class LinkBuffer;
 class PCToCodeOriginMapBuilder;
