--- target-org/webkit/CMakeLists.txt	2021-03-06 03:12:45.000000000 +0800
+++ target/webkit/CMakeLists.txt	2022-07-09 18:13:01.000000000 +0800
@@ -38,9 +38,9 @@
     add_subdirectory(Tools)
 endif ()
 
-if (DEVELOPER_MODE)
-    add_subdirectory(PerformanceTests)
-endif ()
+# if (DEVELOPER_MODE)
+#     add_subdirectory(PerformanceTests)
+# endif ()
 
 # -----------------------------------------------------------------------------
 # Print the features list last, for maximum visibility.
--- target-org/webkit/Source/cmake/OptionsJSCOnly.cmake	2022-07-09 15:31:32.000000000 +0800
+++ target/webkit/Source/cmake/OptionsJSCOnly.cmake	2022-07-09 15:29:24.000000000 +0800
@@ -47,7 +47,7 @@
 if (WIN32)
     set(ENABLE_API_TESTS OFF)
 else ()
-    set(ENABLE_API_TESTS ON)
+    set(ENABLE_API_TESTS OFF)
 endif ()
 
 if (WTF_CPU_ARM OR WTF_CPU_MIPS)
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2022-07-09 15:31:21.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2022-07-09 15:31:50.000000000 +0800
@@ -1564,4 +1565,4 @@
         VERBATIM)
 endif ()
 
-add_subdirectory(shell)
+# add_subdirectory(shell)
