diff -aur target-org/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp target/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp
--- target-org/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp	2019-06-18 21:49:52.000000000 +0800
+++ target/webkit/Source/WTF/wtf/unicode/CollatorDefault.cpp	2019-06-18 22:44:39.000000000 +0800
@@ -28,12 +28,13 @@

 #include "config.h"
 #include <wtf/unicode/Collator.h>
+#include <StringView.h>

 #if UCONFIG_NO_COLLATION

 namespace WTF {

-int Collator::collate(StringView a, StringView b) const
+int Collator::collate(StringView a, StringView b)
 {
     unsigned commonLength = std::min(a.length(), b.length());
     for (unsigned i = 0; i < commonLength; ++i) {
@@ -51,7 +52,7 @@
     return 0;
 }

-int Collator::collateUTF8(const char* a, const char* b) const
+int Collator::collateUTF8(const char* a, const char* b)
 {
     return collate(String::fromUTF8(a), String::fromUTF8(b));
 }
