diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/Source/JavaScriptCore/CMakeLists.txt
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2019-08-21 08:24:27.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2019-09-17 13:54:19.000000000 +0800
@@ -118,6 +118,8 @@
     WebKit::WTF
 )

+list(APPEND JavaScriptCore_LIBRARIES ${ICU_LIBRARIES} ${ICU_DATA_LIBRARIES})
+
 if (USE_CAPSTONE)
     list(APPEND JavaScriptCore_LIBRARIES capstone)
 endif ()
