diff -aur target-org/webkit/Source/JavaScriptCore/API/JSBase.cpp target/webkit/Source/JavaScriptCore/API/JSBase.cpp
--- target-org/webkit/Source/JavaScriptCore/API/JSBase.cpp	2017-05-05 02:37:52.000000000 +0300
+++ target/webkit/Source/JavaScriptCore/API/JSBase.cpp	2018-10-09 14:38:30.000000000 +0300
@@ -197,3 +197,9 @@
 const char iosInstallName60 = 0;
 const char iosInstallName61 = 0;
 #endif
+
+extern "C" int __android_log_print(int prio, const char *tag, const char *fmt, ...);
+__attribute__((constructor))
+void printVersion() {
+    __android_log_print(3, "JavaScriptCore.Version", "%s", JSC_VERSION);
+}
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt      2019-03-26 13:04:34.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt  2019-03-26 13:01:41.000000000 +0800
@@ -122,7 +122,9 @@
     ${ICU_I18N_LIBRARIES}
     ${ICU_LIBRARIES}
     ${ICU_DATA_LIBRARIES}
+    log
 )
+add_definitions(-DJSC_VERSION="${JSC_VERSION}")

 if (USE_CAPSTONE)
     list(APPEND JavaScriptCore_LIBRARIES capstone)
