diff -aur target-org/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp target/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp
--- target-org/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp	2019-06-18 21:49:53.000000000 +0800
+++ target/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp	2019-06-18 22:44:39.000000000 +0800
@@ -23,6 +23,10 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */

+// CUSTOMIZE_REACT_NATIVE: getline() does not implemented until Android NDK API 18.
+// Since MemoryFootprint does not being used in JSC, comment out to prevent build break.
+#if !defined(CUSTOMIZE_REACT_NATIVE)
+
 #include "config.h"
 #include <wtf/MemoryFootprint.h>

@@ -100,3 +104,5 @@
 }

 } // namespace WTF
+
+#endif // !defined(CUSTOMIZE_REACT_NATIVE)
