--- target-org/webkit/Source/WTF/wtf/URLHelpers.cpp	2023-02-04 00:50:26
+++ target/webkit/Source/WTF/wtf/URLHelpers.cpp	2023-02-04 00:50:29
@@ -301,7 +301,7 @@
     Optional<UChar32> previousCodePoint;
     while (i < length) {
         UChar32 c;
-        U16_NEXT(buffer, i, length, c)
+        U16_NEXT(buffer, i, length, c);
         UErrorCode error = U_ZERO_ERROR;
         UScriptCode script = uscript_getScript(c, &error);
         if (error != U_ZERO_ERROR) {
