diff -aur target-icucollator/webkit/Source/WTF/wtf/CMakeLists.txt target/webkit/Source/WTF/wtf/CMakeLists.txt
--- target-icucollator/webkit/Source/WTF/wtf/CMakeLists.txt	2017-07-11 13:16:20.800299018 +0200
+++ target/webkit/Source/WTF/wtf/CMakeLists.txt	2017-07-11 13:17:20.743222116 +0200
@@ -331,9 +331,8 @@
 endif ()
 
 list(APPEND WTF_SOURCES
     unicode/CollatorDefault.cpp

-    unicode/icu/CollatorICU.cpp
 )
 set(WTF_SYSTEM_INCLUDE_DIRECTORIES
     ${ICU_INCLUDE_DIRS}
