diff -aur target-org/webkit/Source/WTF/wtf/CMakeLists.txt target/webkit/Source/WTF/wtf/CMakeLists.txt
--- target-org/webkit/Source/WTF/wtf/CMakeLists.txt	2019-03-26 11:25:32.000000000 +0800
+++ target/webkit/Source/WTF/wtf/CMakeLists.txt	2019-03-26 13:09:15.000000000 +0800
@@ -475,7 +475,6 @@
 list(APPEND WTF_SOURCES
     unicode/CollatorDefault.cpp

-    unicode/icu/CollatorICU.cpp
 )
 set(WTF_SYSTEM_INCLUDE_DIRECTORIES
     ${ICU_INCLUDE_DIRS}
